thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/AnomalyResource.java

```diff
@@ -523,6 +526,9 @@ public Response updateAnomalyFunction(@NotNull @PathParam("id") Long id,
       anomalyFunctionSpec.setFilters(filterString);
     }
     if (StringUtils.isNotBlank(properties)) {
+      if (isOverrideProp) {
```

Comment:
Move this down to before line 534, such that the 'update parts' are  adjacent to each other